src/backend/api/meals.js

```diff
@@ -1,15 +1,123 @@
 const express = require("express");
+const { limit, sum } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    let meals = await knex("meal");
+
+    //Returns all meals that are cheaper than maxPrice.
+    if(request.query.maxPrice !== undefined) {
+      const reqPrice = Number(request.query.maxPrice);
+      if(isNaN(reqPrice)) {
+        response.status(400).send({"message": "MaxPrice should be a number"});
+      } else {
+        meals = await knex("meal").where('price', '<', reqPrice);
```

Comment:
The approach here and with the other query parameter further down does not look like it will work. Instead `meals` will just be assigned to the results from the last used query parameter.